Tests/FluentKitTests/QueryBuilderTests.swift

```diff
@@ -68,4 +68,19 @@ final class QueryBuilderTests: XCTestCase {
         XCTAssertEqual(retrievedPlanets.count, planets.count)
         XCTAssertEqual(retrievedPlanets.map(\.name), planets.map(\.name))
     }
+
+    func testQueryHistory() throws {
+        let starId = UUID()
+        let planets = [
+            Planet(id: UUID(), name: "P1", starId: starId),
+            Planet(id: UUID(), name: "P2", starId: starId),
+            Planet(id: UUID(), name: "P3", starId: starId)
+        ]
+        let test = ArrayTestDatabase()
+        test.append(planets.map(TestOutput.init))
+
+        let retrievedPlanets = try Planet.query(on: test.db).all().wait()
```

Comment:
`test.db` is a computed property that generates a new `Database` (and query history). Try doing `let db = test.db` first.